File: accounts-drive/go/keccak.go
package drive

import "golang.org/x/crypto/sha3"

// Keccak256 hashes the concatenation of the given byte slices with the
// legacy (pre-FIPS, 0x01-padded) Keccak-256 — the hash of the machine's
// Merkle tree and of Ethereum.
//
// golang.org/x/crypto is the de-facto standard implementation; go-ethereum's
// own hashing is built on the same code. Correctness here is still pinned by
// the spec's test vectors, which were generated independently.
func Keccak256(data ...[]byte) [32]byte {
	h := sha3.NewLegacyKeccak256()
	for _, d := range data {
		h.Write(d)
	}
	var out [32]byte
	h.Sum(out[:0])
	return out
}

File: accounts-drive/js/src/keccak.ts
// Keccak-256 (Ethereum's, pre-FIPS 0x01 domain padding) via @noble/hashes —
// the audited, pure-JS de-facto standard (the implementation under ethers and
// viem). Correctness is still pinned by the spec's test vectors, which were
// generated independently.
import { keccak_256 } from "@noble/hashes/sha3.js";

// keccak256 hashes the concatenation of the given byte chunks.
export function keccak256(...data: Uint8Array[]): Uint8Array {
  const h = keccak_256.create();
  for (const d of data) h.update(d);
  return h.digest();
}
